etna/models/seasonal_ma.py

```diff
@@ -81,10 +81,27 @@ def _validate_context(self, df: pd.DataFrame, prediction_size: int):
                 "Given context isn't big enough, try to decrease context_size, prediction_size or increase length of given dataframe!"
             )
 
+    def _predict_components(self, df: pd.DataFrame, prediction_size: int) -> pd.DataFrame:
+        """Estimate forecast components."""
+        from etna.transforms import LagTransform
+
+        self._validate_context(df=df, prediction_size=prediction_size)
+
+        lag_transform = LagTransform(
+            in_column="target",
+            lags=list(range(self.seasonality, self.context_size + 1, self.seasonality)),
+            out_column="target_component_lag",
+        )
+        target_components_df = lag_transform._transform(df) / self.window
```

Comment:
Are we sure that we won't remove `df.copy()` from `LagTransform` in the future?

Reply:
We actually should, now it looks like a bug

Reply:
Fixed all the places with redundant `copy`